File: Problemset/0055-Jump Game/Greedy.rb
# @param {Integer[]} nums
# @return {Boolean}
def can_jump(nums)
  max_index = 0

  (0...nums.size).each do |i|
    return false if i > max_index

    max_index = [max_index, i + nums[i]].max
  end

  true
end

File: Problemset/0055-Jump Game/Greedy.rs
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut max_index = 0;

        for i in 0..nums.len() {
            if i > max_index as usize {
                return false;
            }
            max_index = max_index.max(i as i32 + nums[i]);
        }

        true
    }
}
